--- a/Makefile
+++ b/Makefile
@@ -1,37 +1,38 @@
 CC		?= cc
+AR		?= ar
 RM		?= rm -rf
 INSTALL		?= install
 INSTALL_PROGRAM	?= $(INSTALL)
 INSTALL_DATA	?= $(INSTALL) -m 644
-LUA_V		?= 5.4
-LUA_LDIR	?= /usr/share/lua/$(LUA_V)
-LUA_CDIR	?= /usr/lib/lua/$(LUA_V)
+LUA_LDIR	?= /usr/share/lua
+LUA_CDIR	?= /usr/lib/lua
 T		= lxp
-LIBNAME		= $(T).so
 
 COMMON_CFLAGS	 = -g -pedantic -Wall -O2 -fPIC -DPIC
-LUA_INC		?= -I/usr/include/lua$(LUA_V)
+LUA_INC		?= -I/usr/include/lua
 EXPAT_INC	?= -I/usr/include
 CF		 = $(LUA_INC) $(EXPAT_INC) $(COMMON_CFLAGS) $(CFLAGS)
 
 EXPAT_LIB	 = -lexpat
-COMMON_LDFLAGS	 = -shared
+COMMON_LDFLAGS	 = -llua
 LF		 = $(COMMON_LDFLAGS) $(EXPAT_LIB) $(LDFLAGS)
 
-OBJS		 = src/lxplib.o
+lib: src/$(T)lib.o
 
-lib: src/$(LIBNAME)
-
-src/$(LIBNAME):
+src/$(T)lib.o:
 	export MACOSX_DEPLOYMENT_TARGET="10.3";
-	$(CC) $(CF) -o $@ src/$(T)lib.c $(LF)
+	$(CC) $(CF) -o src/$(T)lib.o -c src/$(T)lib.c
+	$(CC) $(CF) -shared -o src/$(T).so src/$(T)lib.o $(LF)
+	$(AR) rcs src/$(T).a src/$(T)lib.o
 
 install:
-	$(INSTALL_PROGRAM) -D src/$(LIBNAME) $(DESTDIR)$(LUA_CDIR)/$(LIBNAME)
+	$(INSTALL_PROGRAM) -D src/$(T).a $(DESTDIR)$(LUA_CDIR)/$(T).a
+	$(INSTALL_PROGRAM) -D src/$(T).so $(DESTDIR)$(LUA_CDIR)/$(T).so
 	$(INSTALL_PROGRAM) -D src/$T/lom.lua $(DESTDIR)$(LUA_LDIR)/$T/lom.lua
 	$(INSTALL_PROGRAM) -D src/$T/totable.lua $(DESTDIR)$(LUA_LDIR)/$T/totable.lua
 	$(INSTALL_PROGRAM) -D src/$T/threat.lua $(DESTDIR)$(LUA_LDIR)/$T/threat.lua
 
 clean:
-	$(RM) src/$(LIBNAME) $(OBJS)
-	$(RM) ./$(LIBNAME)
+	$(RM) src/$(T).o
+	$(RM) src/$(T).so
+	$(RM) src/$(T).a
