--- a/xinstall.c
+++ b/xinstall.c
@@ -12,6 +12,7 @@
 
 #include "util.h"
 
+static int Sflag = 0;
 static int Dflag = 0;
 static gid_t group;
 static uid_t owner;
@@ -43,6 +44,26 @@
 static int
 install(const char *s1, const char *s2, int depth)
 {
+	if (Sflag) {
+		pid_t          pid; 
+		FILE           *fp; 
+		int            status; 
+		unsigned char  buffer[4]; 
+ 
+		fp = fopen(s1, "rb"); 
+		fread(buffer, sizeof(*buffer), sizeof(buffer) / sizeof(buffer[0]), fp); 
+		fclose(fp);
+
+		if ((buffer[1] == 'E' && buffer[2] == 'L' && buffer[3] == 'F') || 
+		    (buffer[1] == '<' && buffer[2] == 'a' && buffer[3] == 'r')) 
+			switch (pid = fork()) { 
+			case 0: 
+				execl("/usr/bin/strip", "strip", "--strip-unneeded", s1, NULL); 
+			default: 
+				wait(&status); 
+			} 
+	}
+
 	int f1, f2;
 
 	if ((f1 = open(s1, O_RDONLY)) < 0)
@@ -77,7 +98,7 @@
 static void
 usage(void)
 {
-	eprintf("usage: %s [-g group] [-o owner] [-m mode] (-d dir ... | [-D] (-t dest source ... | source ... dest))\n", argv0);
+	eprintf("usage: %s [-s] [-g group] [-o owner] [-m mode] (-d dir ... | [-D] (-t dest source ... | source ... dest))\n", argv0);
 }
 
 int
@@ -104,7 +125,7 @@
 		Dflag = 1;
 		break;
 	case 's':
-		/* no-op for compatibility */
+		Sflag = 1;
 		break;
 	case 'g':
 		gflag = EARGF(usage());
--- a/xinstall.1
+++ b/xinstall.1
@@ -72,6 +72,8 @@
 .Fl m
 is used to select another mode.
 When the symbolic notation is used, the base mode is 0000.
+.It Fl s
+Strip files (if needed) before installing.
 .El
 .Sh SEE ALSO
 .Xr chmod 1 ,
