--- a/meson.build
+++ b/meson.build
@@ -189,9 +189,6 @@
     else
       # -fpic to force dynamic tls, otherwise TLS relaxation defeats check
       gnu2_test = cc.run('int __thread x; int main() { return x; }', args: ['-mtls-dialect=gnu2', '-fpic'], name: '-mtls-dialect=gnu2')
-      if gnu2_test.returncode() == 0
-        add_project_arguments('-mtls-dialect=gnu2', language : ['c'])
-      endif
     endif
   endif
 endif
