--- a/lib/init/rc.boot
+++ b/lib/init/rc.boot
@@ -4,7 +4,7 @@
 # Shared code between boot/shutdown.
 . /usr/lib/init/rc.lib
 
-log "Welcome to KISS!"
+log "Welcome to ???"
 
 log "Mounting pseudo filesystems..."; {
     mnt nosuid,noexec,nodev    proc     proc /proc
@@ -82,8 +82,7 @@
 }
 
 log "Setting hostname..."; {
-    read -r hostname < /etc/hostname
-    printf %s "${hostname:-KISS}" > /proc/sys/kernel/hostname
+    cat /etc/hostname > /proc/sys/kernel/hostname
 } 2>/dev/null
 
 log "Loading sysctl settings..."; {
--- a/lib/init/rc.lib
+++ b/lib/init/rc.lib
@@ -5,7 +5,7 @@
 # the rc.boot and rc.shutdown scripts.
 
 log() {
-    printf '\033[31;1m=>\033[m %s\n' "$@"
+    printf '\033[31;1m@\033[m %s\n' "$@"
 
     # Only print to /dev/kmsg and /dev/null if they both exist.
     # This will fail late in the shutdown process and possibly
