--- a/build.rs
+++ b/build.rs
@@ -117,6 +117,8 @@ fn main() {
         .include("curl/lib")
         .include("curl/include")
         .define("BUILDING_LIBCURL", None)
+        .define("CURL_CA_BUNDLE", "\"/etc/ssl/cert.pem\"")
+        .define("CURL_CA_PATH", "\"/etc/ssl/certs\"")
         .define("CURL_DISABLE_DICT", None)
         .define("CURL_DISABLE_GOPHER", None)
         .define("CURL_DISABLE_IMAP", None)
