--- a/meson.build.old
+++ b/meson.build
@@ -93,11 +93,11 @@
   INTERFACE_INCLUDE_DIRECTORIES "''' + join_paths('${PACKAGE_PREFIX_DIR}', get_option('includedir')) + '")')
 endif
 
-import('cmake').configure_package_config_file(
-  name: 'jsoncpp',
-  input: 'jsoncppConfig.cmake.meson.in',
-  configuration: cmakeconf)
-install_data('jsoncpp-namespaced-targets.cmake', install_dir : join_paths(get_option('libdir'), 'cmake', jsoncpp_lib.name()))
+#import('cmake').configure_package_config_file(
+#  name: 'jsoncpp',
+#  input: 'jsoncppConfig.cmake.meson.in',
+#  configuration: cmakeconf)
+#install_data('jsoncpp-namespaced-targets.cmake', install_dir : join_paths(get_option('libdir'), 'cmake', jsoncpp_lib.name()))
 
 # for libraries bundling jsoncpp
 jsoncpp_dep = declare_dependency(
