--- a/pathexec_run.c
+++ b/pathexec_run.c
@@ -1,5 +1,6 @@
 /* Public domain. */
 
+#include <unistd.h>
 #include "error.h"
 #include "stralloc.h"
 #include "str.h"
--- a/seek_set.c
+++ b/seek_set.c
@@ -1,5 +1,6 @@
 /* Public domain. */
 
+#include <unistd.h>
 #include <sys/types.h>
 #include "seek.h"
 
--- a/prot.c
+++ b/prot.c
@@ -1,5 +1,6 @@
 /* Public domain. */
 
+#include <unistd.h>
 #include "hasshsgr.h"
 #include "prot.h"
 
--- a/sig_block.c
+++ b/sig_block.c
@@ -4,6 +4,8 @@
 #include "sig.h"
 #include "hassgprm.h"
 
+#define HASSIGPROCMASK 1
+
 void sig_block(int sig)
 {
 #ifdef HASSIGPROCMASK
